c++/src/kj/compat/http.c++

```diff
@@ -1045,6 +1045,65 @@ kj::String HttpHeaders::toString() const {
   return serialize(nullptr, nullptr, nullptr, nullptr);
 }
 
+// TODO (now): Where should we put these 2 functions?
+// I've put them here for convenience, but this should only be temporary.
+// We should also write tests.
+inline kj::Maybe<StringPtr> advanceToNext(const StringPtr input, const char& c,
+    kj::Maybe<size_t>& advancedBy) {
+  // Given a string and character, return a StringPtr to the first non-whitespace character
+  // following the first instance of `char c`. The index of `char c` relative to `input` is stored
+  // in `advancedBy`.
+  kj::Maybe<StringPtr> result;
+  advancedBy = input.findFirst(c);
+  KJ_IF_MAYBE(index, advancedBy) {
+    // We found a delimiter, move to 1 past it, then skip over the whitespace.
+    auto startOfNext = const_cast<char*>((input.begin() + *index + 1));
+    result = skipSpace(startOfNext);
+  }
+  return result;
+}
+
+kj::Vector<kj::String> splitPartsWithCondition(const kj::StringPtr input, const char& delim,
```

Comment:
To give some context for why I've included the ability to skip if a precondition isn't met: when we're parsing the headers as a client (prior to sending the request to the server), we only want to retain `permessage-deflate` offers, so we might as well do this in 1 pass.

Reply:
Additionally, `kj/compat/url.c++` has a few `split` variants, but extensions have the abnf 
```
Sec-WebSocket-Extensions = extension-list
         extension-list = 1#extension
         extension = extension-token *( ";" extension-param )
```
so this is a valid `extension-list`: `,,,          ,permessage-deflate;client_max_window_bits`. Since we can have 1+ commas after an extension, as well as optional linear whitespace, it seemed easier to have a function that just gives us what we care about so we can focus on processing that.